specifyweb/frontend/js_src/lib/components/AttachmentsBulkImport/Import.tsx

```diff
@@ -0,0 +1,368 @@
+import React from 'react';
+import { useParams } from 'react-router-dom';
+
+import { usePromise } from '../../hooks/useAsyncState';
+import { attachmentsText } from '../../localization/attachments';
+import { commonText } from '../../localization/common';
+import { wbText } from '../../localization/workbench';
+import { ajax } from '../../utils/ajax';
+import { f } from '../../utils/functools';
+import type { RA } from '../../utils/types';
+import { Container } from '../Atoms';
+import { Button } from '../Atoms/Button';
+import type { UiFormatter } from '../Forms/uiFormatters';
+import { Dialog, LoadingScreen } from '../Molecules/Dialog';
+import { FilePicker } from '../Molecules/FilePicker';
+import { TableIcon } from '../Molecules/TableIcon';
+import { NotFoundView } from '../Router/NotFoundView';
+import { SafeRollbackAttachmentsNew } from './AttachmentsRollback';
+import { SafeUploadAttachmentsNew } from './AttachmentsUpload';
+import { AttachmentsValidationDialog } from './AttachmentsValidationDialog';
+import { fetchAttachmentResourceId } from './fetchAttachmentResource';
+import { staticAttachmentImportPaths } from './importPaths';
+import { RenameAttachmentDataSetDialog } from './RenameAttachmentDataSet';
+import { SelectUploadPath } from './SelectUploadPath';
+import type {
+  AttachmentDataSetResource,
+  FetchedDataSet,
+  PartialUploadableFileSpec,
+  SavedAttachmentDataSetResource,
+  UnBoundFile,
+} from './types';
+import { useEagerDataSet } from './useEagerDataset';
+import {
+  matchSelectedFiles,
+  reconstructDeletingAttachment,
+  reconstructUploadingAttachmentSpec,
+  resolveFileNames,
+} from './utils';
+import { ViewAttachmentFiles } from './ViewAttachmentFiles';
+
+export type AttachmentUploadSpec = {
+  readonly staticPathKey: keyof typeof staticAttachmentImportPaths;
+  readonly formatQueryResults: (
+    value: number | string | null | undefined
+  ) => string | undefined;
+};
+export type PartialAttachmentUploadSpec = {
+  readonly fieldFormatter?: UiFormatter;
+} & (AttachmentUploadSpec | { readonly staticPathKey: undefined });
+
+type AttachmentDataSet =
+  | AttachmentDataSetResource
+  | SavedAttachmentDataSetResource;
+export type EagerDataSet = AttachmentDataSet & {
+  readonly needsSaved: boolean;
+  readonly save: boolean;
+};
+
+const newAttachmentDataSetResource: AttachmentDataSetResource = {
+  name: attachmentsText.newAttachmentDataset(),
+  uploadableFiles: [],
+  uploadSpec: { staticPathKey: undefined },
+  status: 'main',
+};
+export function NewAttachmentImport(): JSX.Element | null {
```

Comment:
the way you have it is probably better for familiarity, but a FYI that this also works (and is faster, but not going to be noticeable):
```ts
const NewAttachmentImport = <AttachmentsImport
      attachmentDataSetResource={newAttachmentDataSetResource}
    />;
```

Reply:
We do this for `loadingBar` in a few places in the codebase